### PR TITLE
fix all docs.substrate.io links to use new site 

### PR DESCRIPTION
**Pull request by nuke-web3**
https://github.com/substrate-developer-hub/substrate-docs/issues/1132

NOTE: SS58 page is not active at this time, so links to that reference are removed. https://github.com/paritytech/ss58-registry/issues/112

### CODE REVIEW

File: docs/build/build-node-interaction.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ This page will guide you through some basic interactions with your node. This gu
 seen as canonical reference. Always refer to the proper
 documentation for the tool you are using:
 
-- [Substrate RPC API](https://docs.substrate.io/rustdocs/latest/sc_rpc_api/index.html)
+- [Substrate RPC API](https://paritytech.github.io/substrate/master/sc_rpc_api/index.html)
```
Comment (profradha) [inline docs/build/build-node-interaction.md:14]: Why is this no longer listed on https://docs.substrate.io/ ?

File: docs/build/build-protocol-info.md [hunk 2]
```diff
@@ -41,16 +41,15 @@ Polkadot (and Substrate) use the SS58 address format. This is a broad "meta-form
...
+See the [SS58 page](https://docs.substrate.io/main-docs/fundamentals/accounts-addresses-keys/#address-encoding-and-chain-specific-addresses) in the Substrate documentation for
 encoding information and a more comprehensive list of network prefixes.
 
 :::warning Do not use regular expressions (regex) to validate addresses
 
 Always verify using the
 prefix and checksum of the address. Substrate API Sidecar provides an
 `accounts/{accountId}/validate` path that returns a boolean `isValid` response for a provided
-address. If you want to validate addresses in another way, see the
-[validation documentation](https://docs.substrate.io/v3/advanced/ss58/#validating-addresses).
```
Comment (profradha) [inline docs/build/build-protocol-info.md:53]: Are the alternative ways to validate addresses no longer listed on Substrate docs?

File: docs/learn/learn-treasury.md [hunk 3]
```diff
@@ -330,4 +330,4 @@ migration to a minority fork. However, the possibility of this scenario is quite
 
 - [Substrate's Treasury Pallet](https://github.com/paritytech/substrate/blob/master/frame/treasury/src/lib.rs) -
   The Rust implementation of the Treasury.
-  ([Docs](https://docs.substrate.io/rustdocs/latest/pallet_treasury/index.html))
+  ([Docs](https://paritytech.github.io/substrate/master/pallet_treasury/index.html))
```
Comment (profradha) [inline docs/learn/learn-treasury.md:333]: So, all the rustdocs now live on https://paritytech.github.io/substrate/ ?

File: docs/build/build-ss58-registry.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ description: Information about SS58 encoding.
 slug: ../build-ss58-registry
 ---
 
-:::note Substrate-based chains use an [SS58 encoding](https://docs.substrate.io/v3/advanced/ss58/) for their address formats.
+:::note Substrate-based chains use an SS58 encoding for their address formats.
```
Comment (profradha) [inline docs/build/build-ss58-registry.md:9]: Are there plans to add a dedicated page on SS58 encoding? If not, we can link it to https://docs.substrate.io/main-docs/fundamentals/accounts-addresses-keys/#address-encoding-and-chain-specific-addresses

**Review by DrW3RK — APPROVED** [review]
I've checked the links added and all of them are live. Have a few questions, but they are not blockers for this PR. Good to go!

File: docs/build/build-ss58-registry.md (hunk 4 above)
Comment (profradha) [reply docs/build/build-ss58-registry.md:9]: Just saw the issue you linked

File: docs/learn/learn-treasury.md (hunk 3 above)
Comment (nuke-web3) [reply docs/learn/learn-treasury.md:333]: yes, specifically master. Others will be deprecated.

Comment (nuke-web3) [reply docs/learn/learn-treasury.md:333]: hope to be resolved soon, I will make a PR here when I have a location for it

File: docs/build/build-node-interaction.md (hunk 1 above)
Comment (nuke-web3) [reply docs/build/build-node-interaction.md:14]: no

**Pull request merged**